File: frontend/src/api/passHistory.ts
import { api } from "./client";

export type PassHistoryItem = {
  id: number;
  pass_id: number;
  ride_id?: number | null;
  operation: "DEDUCT" | "RESTORE";
  entries: number;
  note?: string | null;
  created_at: string;
  ride_date?: string | null;
  ride_start_time?: string | null;
  horse_name?: string | null;
  client_name?: string | null;
  instructor_name?: string | null;
};

export async function getPassHistory(
  passId: number
): Promise<PassHistoryItem[]> {
  const { data } = await api.get<PassHistoryItem[]>(
    `/pass-history/${passId}`
  );

  return data;
}

File: backend/app/schemas/pass_history.py
from datetime import datetime

from pydantic import BaseModel, ConfigDict


class PassHistoryRead(BaseModel):
    id: int
    pass_id: int
    ride_id: int | None

    operation: str
    entries: int
    note: str | None

    created_at: datetime

    ride_date: str | None
    ride_start_time: str | None

    horse_name: str | None
    client_name: str | None
    instructor_name: str | None

    model_config = ConfigDict(from_attributes=True)